control/freqplot.py

```diff
@@ -1,141 +1,142 @@
 
-__all__ = ['bode_plot', 'nyquist_plot', 'gangof4_plot', 'singular_values_plot',
+__all__ = ['bode_plot', 'NyquistResponseData', 'nyquist_response',
+           'nyquist_plot', 'singular_values_response',
+           'singular_values_plot', 'gangof4_plot', 'gangof4_response',
            'bode', 'nyquist', 'gangof4']
 
+# Default font dictionary
+_freqplot_rcParams = mpl.rcParams.copy()
+_freqplot_rcParams.update({
+    'axes.labelsize': 'small',
+    'axes.titlesize': 'small',
+    'figure.titlesize': 'medium',
+    'legend.fontsize': 'x-small',
+    'xtick.labelsize': 'small',
+    'ytick.labelsize': 'small',
+})
+
 # Default values for module parameter variables
 _freqplot_defaults = {
+    'freqplot.rcParams': _freqplot_rcParams,
     'freqplot.feature_periphery_decades': 1,
     'freqplot.number_of_samples': 1000,
     'freqplot.dB': False,  # Plot gain in dB
     'freqplot.deg': True,  # Plot phase in degrees
     'freqplot.Hz': False,  # Plot frequency in Hertz
     'freqplot.grid': True,  # Turn on grid for gain and phase
     'freqplot.wrap_phase': False,  # Wrap the phase plot at a given value
-
-    # deprecations
-    'deprecated.bode.dB': 'freqplot.dB',
-    'deprecated.bode.deg': 'freqplot.deg',
-    'deprecated.bode.Hz': 'freqplot.Hz',
-    'deprecated.bode.grid': 'freqplot.grid',
-    'deprecated.bode.wrap_phase': 'freqplot.wrap_phase',
+    'freqplot.freq_label': "Frequency [%s]",
+    'freqplot.share_magnitude': 'row',
+    'freqplot.share_phase': 'row',
+    'freqplot.share_frequency': 'col',
 }
 
-
 #
-# Main plotting functions
+# Frequency response data list class
 #
-# This section of the code contains the functions for generating
-# frequency domain plots
+# This class is a subclass of list that adds a plot() method, enabling
+# direct plotting from routines returning a list of FrequencyResponseData
+# objects.
 #
 
+class FrequencyResponseList(list):
+    def plot(self, *args, plot_type=None, **kwargs):
+        if plot_type == None:
+            for response in self:
+                if plot_type is not None and response.plot_type != plot_type:
+                    raise TypeError(
+                        "inconsistent plot_types in data; set plot_type "
+                        "to 'bode', 'nichols', or 'svplot'")
+                plot_type = response.plot_type
+
+        # Use FRD plot method, which can handle lists via plot functions
+        return FrequencyResponseData.plot(
+            self, plot_type=plot_type, *args, **kwargs)
+
 #
 # Bode plot
 #
+# This is the default method for plotting frequency responses.  There are
+# lots of options available for tuning the format of the plot, (hopefully)
+# covering most of the common use cases.
+#
 
-
-def bode_plot(syslist, omega=None,
-              plot=True, omega_limits=None, omega_num=None,
-              margins=None, method='best', *args, **kwargs):
+def bode_plot(
+        data, omega=None, *fmt, ax=None, omega_limits=None, omega_num=None,
```

Comment:
This seems to be the time to deprecate `omega`, `omega_limits`, and `omega_num` in favor of the more general `frequency`,`frequency_limits`, and `frequency_num` given that omega suggests rad/s even though these are specified in either rad/s or hz depending on the defaults.

Reply:
Edit: looking elsewhere, sticking with `omega` may make more sense for nyquist and nichols. not sure about bode yet. Would be nice to be able to specify ranges in Hz for the bode plot though - was this a feature before?